include/algorithms/synthesis/dd_synthesis.hpp

```diff
@@ -13,9 +13,9 @@ namespace syrec {
 
     class DDSynthesizer {
     public:
-        static auto synthesize(const TruthTable& tt, const bool onePass = false) -> std::shared_ptr<qc::QuantumComputation> {
+        static auto synthesize(const TruthTable& tt, const bool encodeWithoutAdditionalLine = true, const bool onePass = false) -> std::shared_ptr<qc::QuantumComputation> {
```

Comment:
It does not really make sense to keep adding options to this function, e.g., the combination `encodeWithoutAdditionalLine = true` and  `const bool onePass = true` does not really make sense or rather the first option does not have an effect on the one pass synthesis.

I feel that it would be better to clearly split the synthesis techniques, e.g., 
 - a generic `synthesize` function that neither uses coding techniques nor the one-pass synthesis, 
 - a `synthesizeCodingTechniques` function that allows to specify the corresponding option you want to add here
 - a `synthesizeOnePass` function that realises the one-pass synthesis.

this creates a separation of concerns and should stop the addition of more and more options to a single function.
furthermore it encapsulates the functionality in a better fashion and does not clobber everything together into single functions. Many of the functions right now consist of multiple huge `if ... else` blocks that handle the respective synthesis. This is not maintainable in the future and should be refactored. This addition here is the perfect opportunity to do that in order to get a feeling of how easy it is to extend the existing synthesis techniques.

Of course, it is key to share as much code as possible and avoid code duplication. One possible strategy for that could be to realize the above solution by implementing common functionality in the `DDSynthesizer` class and then deriving from that class for the coding techniques and the one-pass technique (only specialising those functions that need to be adapted).